imports/plugins/included/product-detail-simple/client/components/productDetail.js

```diff
@@ -81,6 +102,176 @@ class ProductDetail extends Component {
     return null;
   }
 
+  switchDigital(e) {
+    this.setState({digital: e.target.checked});
+    const productId = this.props.product._id;
+    const checked = e.target.checked;
+    this.props.onProductFieldChange(productId, "isDigital", checked);
+    this.props.changeParentIsDigitalState(e.target.checked);
+    if (!this.state.digital) {
+      this.setState({cartQuantity: "hidden"});
+    } else {
+      this.setState({cartQuantity: "number"});
+    }
+  }
+
+  renderInputToggle() {
+    if (this.props.hasAdminPermission) {
+      return (
+        <div className="switch-middle">
+          <b className="switch-text">Physical Product</b>
+            <label className="switch">
+              <input id="digital" type="checkbox" defaultChecked={this.state.digital} onChange={this.switchDigital}/>
+              <div className="slider round"/>
+            </label>
+          <b className="switch-text">Digital Product</b>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  handleCategoryStateChange(e) {
+    const value = e.target.value;
+    this.setState({categoryValue: value});
+  }
+
+  handleUploadClick(e) {
+    const files = new FS.File(e.target.files[0]);
+    files.metadata = {
+      productId: this.props.product._id,
+      ownerId: Meteor.userId(),
+      shopId: Reaction.getShopId()
+    };
+
+    if (this.state.categoryValue === "audio") {
+      Audio.insert(files, (err, file) => {
+        if (err) {
+          return Alerts.toast("Something went wrong", "warning");
+        }
+        if (file) {
+          this.setState({fileId: file._id});
+          const digitalInfo = {
+            fileId: file._id,
+            category: this.state.categoryValue
+          };
+
+          this.props.onProductFieldChange(this.props.product._id, "productFileId", file._id);
+          this.props.onProductFieldChange(this.props.product._id, "digitalInfo", digitalInfo);
+          return Alerts.toast("File successfully uploaded", "success");
+        }
+        return null;
+      });
+      return true;
+    }
+
+    if (this.state.categoryValue === "video") {
```

Comment:
All the methods below are repetitive as well. we can create one method and re use it.